stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -1221,7 +1267,16 @@ extension Unicode.Scalar.Properties {
   /// This property corresponds to the "Name" property in the
   /// [Unicode Standard](http://www.unicode.org/versions/latest/).
   public var name: String? {
-    return _scalarName(__swift_stdlib_U_UNICODE_CHAR_NAME)
+    if let fastName = _fastScalarName() {
+      return fastName
+    }
+
+    // The longest name that Unicode defines is 88 characters long.
+    let name = String(_uninitializedCapacity: 90) { buffer in
+      _swift_stdlib_getScalarName(_scalar.value, buffer.baseAddress)
```

Comment:
The maximum name length should be a constant generated by the tool rather than a hard-coded magic number.

Reply:
It can be in the header file, definitely. Either way, a C function that takes a pointer and writes to memory some length beyond that address should probably take a capacity argument and never exceed that.